### PR TITLE
Add TableToTable aggregate coercion support, new IR node

### PR DESCRIPTION
**Pull request by bryancroteau-MSFT**
2 new tests are passing related to the CharT function

### CODE REVIEW

File: src/libraries/Microsoft.PowerFx.Core/Functions/BuiltinFunction.cs [hunk 1]
```diff
@@ -11,13 +11,21 @@ namespace Microsoft.PowerFx.Core.Functions
...
     {
-        public static readonly DName OneColumnTableResultName = new DName("Result");
-        public static readonly DName ColumnName_Name = new DName("Name");
-        public static readonly DName ColumnName_Address = new DName("Address");
-        public static readonly DName ColumnName_Value = new DName("Value");
-        public static readonly DName ColumnName_FullMatch = new DName("FullMatch");
-        public static readonly DName ColumnName_SubMatches = new DName("SubMatches");
-        public static readonly DName ColumnName_StartMatch = new DName("StartMatch");
+        public const string OneColumnTableResultNameStr = "Result";
+        public const string ColumnName_NameStr = "Name";
```
Comment (lesaltzm) [inline src/libraries/Microsoft.PowerFx.Core/Functions/BuiltinFunction.cs:15]: should these literals be private? are they used elsewhere, and what was the benefit of splitting them like this?

Comment (bryancroteau-MSFT) [reply src/libraries/Microsoft.PowerFx.Core/Functions/BuiltinFunction.cs:15]: The literals like Result and Value are used sometimes in the interpreter when reading or creating one-column tables.

File: src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs [hunk 2]
```diff
@@ -261,6 +261,46 @@ public override FormulaValue Visit(UnaryOpNode node, SymbolContext context)
...
+                var resultRows = new List<DValue<RecordValue>>();
+                foreach (var row in table.Rows)
+                {
+                    if (row.IsValue)
+                    {
+                        List<NamedValue> fields = new List<NamedValue>();
+                        foreach (var coercion in node.FieldCoercions)
+                        {
+                            var record = row.Value;
+                            var newScope = context.WithScope(node.Scope).WithScopeValues(record);
```
Comment (lesaltzm) [inline src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs:281]: you can pull `var scopeContext = context.WithScope(node.Scope)` out of the loop, and reuse it.

`var newScope = scopeContext.WithScopeValues(record)`

File: src/libraries/Microsoft.PowerFx.Core/Functions/BuiltinFunction.cs (hunk 1 above)
Comment (lesaltzm) [reply src/libraries/Microsoft.PowerFx.Core/Functions/BuiltinFunction.cs:15]: ah, I see.

**Review by lesaltzm — APPROVED** [review]

File: src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs (hunk 2 above)
Comment (bryancroteau-MSFT) [reply src/libraries/Microsoft.PowerFx.Interpreter/EvalVisitor.cs]: Done

**Review by lesaltzm — APPROVED** [review]

**Pull request merged**